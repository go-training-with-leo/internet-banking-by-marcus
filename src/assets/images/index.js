import { ReactComponent as PlusIcon } from 'assets/images/icons/plus.svg';
import { ReactComponent as CreditCard } from 'assets/images/icons/creditCard.svg';
import { ReactComponent as Contact } from 'assets/images/icons/contact.svg';
import { ReactComponent as CreditCardDone } from 'assets/images/icons/creditCardDone.svg';
import { ReactComponent as Clock } from 'assets/images/icons/time.svg';
import { ReactComponent as DeleteIcon } from 'assets/images/icons/delete.svg';
import { ReactComponent as Edit } from 'assets/images/icons/edit.svg';
import { ReactComponent as Transfer } from 'assets/images/icons/transfer.svg';
import { ReactComponent as Cash } from 'assets/images/icons/cash.svg';
import { ReactComponent as CashAdd } from 'assets/images/icons/cashAdd.svg';
import { ReactComponent as Cancel } from 'assets/images/icons/cancel.svg';
import { ReactComponent as User } from 'assets/images/icons/user.svg';
import { ReactComponent as ArrowDown } from 'assets/images/icons/arrowDown.svg';
import { ReactComponent as Check } from 'assets/images/icons/check.svg';
import { ReactComponent as Reverse } from 'assets/images/icons/reverse.svg';
import { ReactComponent as NotifyBusy } from 'assets/images/icons/notificationBusy.svg';
import { ReactComponent as Info } from 'assets/images/icons/info.svg';
import { ReactComponent as NotifyFree } from 'assets/images/icons/notificationFree.svg';
import { ReactComponent as NotifyRemove } from 'assets/images/icons/notificationRemove.svg';
import { ReactComponent as Alert } from 'assets/images/icons/alert.svg';
import { ReactComponent as AlertYellow } from 'assets/images/icons/alertYellow.svg';
import { ReactComponent as Locker } from 'assets/images/icons/locker.svg';
import { ReactComponent as EightLogo } from 'assets/images/icons/eightBank.svg';
import { ReactComponent as MasterCard } from 'assets/images/icons/masterCard.svg';
import { ReactComponent as Next } from 'assets/images/icons/next.svg';
import { ReactComponent as Back } from 'assets/images/icons/back.svg';
import { ReactComponent as Ellipse } from 'assets/images/icons/ellipse.svg';
import { ReactComponent as Search } from 'assets/images/icons/search.svg';

import { ReactComponent as Loading } from 'assets/images/icons/loadingV2.svg';
import { ReactComponent as Visa } from 'assets/images/icons/visa.svg';
import { ReactComponent as Napas } from 'assets/images/icons/napas.svg';

import Eye from 'assets/images/icons/eye.png';
import EyeCross from 'assets/images/icons/eyeCross.png';

import { ReactComponent as ACB } from 'assets/images/icons/acb.svg';

import EightGif from './eight.gif';

export {
  ACB,
  Alert,
  AlertYellow,
  ArrowDown,
  Back,
  Cancel,
  Cash,
  CashAdd,
  Check,
  Clock,
  Contact,
  CreditCard,
  CreditCardDone,
  DeleteIcon,
  Edit,
  EightGif,
  EightLogo,
  Ellipse,
  Eye,
  EyeCross,
  Info,
  Loading,
  Locker,
  MasterCard,
  Napas,
  Next,
  NotifyBusy,
  NotifyFree,
  NotifyRemove,
  PlusIcon,
  Reverse,
  Search,
  Transfer,
  User,
  Visa,
};
