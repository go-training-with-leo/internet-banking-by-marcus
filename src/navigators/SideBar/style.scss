@import "assets/scss/mixins.scss";
@import "assets/scss/variables.scss";
@import "assets/scss/extends.scss";

.side-bar {
  width: fit-content;
  min-width: 223px;
  height: 100vh;
  display: none;
  &.showSideBar {
    display: flex;
    position: fixed;
    flex-direction: column;
    justify-content: space-between;
    left: 0;
    z-index: 100;
  }
  background-color: $black-medium;

  @include Responsive(SM) {
    @include setFlex(column, space-between, null, null, null);
  }

  &__top {
    @include setFlex(column, null, null, 44px, null);
    .navigate-btn {
      @include setFlex(row, space-between, center, null, null);
      & > .logo {
        margin-top: 22px;
        margin-left: 24px;
      }
      & > .back-btn {
        margin-top: 25px;
        cursor: pointer;
        transition: all 0.3s linear;
        &:hover {
          transform: scale(1.1);
        }
        @include Responsive(SM) {
          display: none;
        }
      }
    }

    & > .list-item {
      @include setFlex(column, null, null, 18px, null);
    }
  }
}
