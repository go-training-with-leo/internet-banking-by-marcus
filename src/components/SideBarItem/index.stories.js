import React from 'react';

import SideBarItem from 'components/SideBarItem';

import { PlusIcon } from 'assets/images';

export default {
  title: 'Menu Item',
  component: SideBarItem,
};

const Default = () => (
  <SideBarItem>
    <PlusIcon /> Label
  </SideBarItem>
);

export { Default };
