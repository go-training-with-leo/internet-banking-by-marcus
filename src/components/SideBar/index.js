import React from 'react';
import PropTypes from 'prop-types';

import { EightLogo } from 'assets/images';

import './style.scss';

const SideBar = ({ children, bottomItem }) => {
  return (
    <div className='side-bar'>
      <div className='side-bar__top'>
        <EightLogo className='logo' />
        <div className='list-item'>{children}</div>
      </div>
      {bottomItem}
    </div>
  );
};

SideBar.defaultProps = {
  children: undefined,
  bottomItem: undefined,
};

SideBar.propTypes = {
  children: PropTypes.node,
  bottomItem: PropTypes.node,
};

export default SideBar;
