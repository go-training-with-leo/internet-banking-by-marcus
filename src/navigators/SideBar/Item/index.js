import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { memo } from 'react';

import './style.scss';

const SideBarItem = ({ isActive, isStatic, children, onClick }) => {
  return (
    <div
      className={classNames('sidebar-item', {
        active: isActive,
        static: isStatic,
      })}
      role='listitem'
      onClick={onClick}
    >
      {children}
    </div>
  );
};

SideBarItem.defaultProps = {
  isActive: false,
  isStatic: false,
  children: undefined,
  onClick: undefined,
};

SideBarItem.propTypes = {
  isActive: PropTypes.bool,
  isStatic: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  onClick: PropTypes.func,
};

export default memo(SideBarItem);
