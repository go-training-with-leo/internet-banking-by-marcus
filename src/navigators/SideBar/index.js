import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { memo } from 'react';

import { Back, EightLogo } from 'assets/images';

import './style.scss';
import useWindowDimensions from 'components/hooks/useWindowDimensions';

const SideBar = ({ children, bottomItem, showSideBar, setShowSideBar }) => {
  const { width } = useWindowDimensions();
  return (
    <div
      className={classNames('side-bar', {
        showSideBar: showSideBar && width < 768,
      })}
    >
      <div className='side-bar__top'>
        <div className='navigate-btn'>
          <EightLogo className='logo' />
          <Back
            className='back-btn'
            width={30}
            height={30}
            fill='white'
            onClick={setShowSideBar}
          />
        </div>
        <div className='list-item'>{children}</div>
      </div>
      {bottomItem}
    </div>
  );
};

SideBar.defaultProps = {
  children: undefined,
  bottomItem: undefined,
};

SideBar.propTypes = {
  children: PropTypes.node,
  bottomItem: PropTypes.node,
};

export default memo(SideBar);
