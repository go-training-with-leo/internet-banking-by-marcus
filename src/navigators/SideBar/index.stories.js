import React from 'react';

import {
  CreditCard,
  Contact,
  Transfer,
  Cash,
  Clock,
  User,
} from 'assets/images';
import SideBarItem from './Item';
import SideBar from '.';

export default {
  title: 'SideBar',
  component: SideBar,
};

const listItem = [
  {
    id: 'cards',
    label: 'Cards',
    icon: <CreditCard width={20} height={20} />,
    isActive: true,
  },
  {
    id: 'contacts',
    label: 'Contacts',
    icon: <Contact width={20} height={20} />,
  },
  {
    id: 'transfer',
    label: 'Transfer',
    icon: <Transfer width={20} height={20} />,
  },
  {
    id: 'debt',
    label: 'Debt',
    icon: <Cash width={20} height={20} />,
  },
  {
    id: 'history',
    label: 'History',
    icon: <img src={Clock} width={20} height={20} alt='clock' />,
  },
];

const Default = () => (
  <SideBar
    bottomItem={
      <SideBarItem>
        <User width={20} height={20} />
        Account
      </SideBarItem>
    }
  >
    {listItem.map(({ id, isActive, label, icon }) => (
      <SideBarItem key={id} isActive={isActive}>
        {icon}
        {label}
      </SideBarItem>
    ))}
  </SideBar>
);

export { Default };
