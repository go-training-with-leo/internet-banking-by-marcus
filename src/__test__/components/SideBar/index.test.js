import React from 'react';
import { render } from '@testing-library/react';
import SideBar from 'navigators/SideBar';
import SideBarItem from 'navigators/SideBar/Item';
import { PlusIcon } from 'assets/images';

test('Test props children', () => {
  const tree = render(
    <SideBar>
      <PlusIcon />
      <SideBarItem>Text</SideBarItem>
    </SideBar>
  );

  const items = tree.container.querySelector('div.list-item');
  expect(items.childElementCount).toEqual(2);
  expect(tree).toMatchSnapshot();
});

test('Test bottomItem props', () => {
  const tree = render(
    <SideBar bottomItem={<SideBarItem>Label</SideBarItem>}>
      <PlusIcon />
      <SideBarItem>Text</SideBarItem>
    </SideBar>
  );

  const sidebar = tree.container.querySelector('div.side-bar');

  expect(sidebar.lastChild).toHaveTextContent('Label');

  expect(tree).toMatchSnapshot();
});
