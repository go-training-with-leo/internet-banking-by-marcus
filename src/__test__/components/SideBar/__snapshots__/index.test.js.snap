// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test bottomItem props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="side-bar"
      >
        <div
          class="side-bar__top"
        >
          <div
            class="navigate-btn"
          >
            <svg
              class="logo"
            >
              eightBank.svg
            </svg>
            <svg
              class="back-btn"
              fill="white"
              height="30"
              width="30"
            >
              back.svg
            </svg>
          </div>
          <div
            class="list-item"
          >
            <svg>
              plus.svg
            </svg>
            <div
              class="sidebar-item"
              role="listitem"
            >
              Text
            </div>
          </div>
        </div>
        <div
          class="sidebar-item"
          role="listitem"
        >
          Label
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="side-bar"
    >
      <div
        class="side-bar__top"
      >
        <div
          class="navigate-btn"
        >
          <svg
            class="logo"
          >
            eightBank.svg
          </svg>
          <svg
            class="back-btn"
            fill="white"
            height="30"
            width="30"
          >
            back.svg
          </svg>
        </div>
        <div
          class="list-item"
        >
          <svg>
            plus.svg
          </svg>
          <div
            class="sidebar-item"
            role="listitem"
          >
            Text
          </div>
        </div>
      </div>
      <div
        class="sidebar-item"
        role="listitem"
      >
        Label
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test props children 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="side-bar"
      >
        <div
          class="side-bar__top"
        >
          <div
            class="navigate-btn"
          >
            <svg
              class="logo"
            >
              eightBank.svg
            </svg>
            <svg
              class="back-btn"
              fill="white"
              height="30"
              width="30"
            >
              back.svg
            </svg>
          </div>
          <div
            class="list-item"
          >
            <svg>
              plus.svg
            </svg>
            <div
              class="sidebar-item"
              role="listitem"
            >
              Text
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="side-bar"
    >
      <div
        class="side-bar__top"
      >
        <div
          class="navigate-btn"
        >
          <svg
            class="logo"
          >
            eightBank.svg
          </svg>
          <svg
            class="back-btn"
            fill="white"
            height="30"
            width="30"
          >
            back.svg
          </svg>
        </div>
        <div
          class="list-item"
        >
          <svg>
            plus.svg
          </svg>
          <div
            class="sidebar-item"
            role="listitem"
          >
            Text
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
